src/model/encoding/convertFromHTMLToContentBlocks.js

```diff
@@ -116,23 +118,76 @@ function getBlockDividerChunk(block: DraftBlockType, depth: number): Chunk {
   };
 }
 
-function getBlockTypeForTag(tag: string, lastList: ?string): DraftBlockType {
-  switch (tag) {
-    case 'h1':
-      return 'header-one';
-    case 'h2':
-      return 'header-two';
-    case 'li':
+function getListBlockType(
+  tag: string,
+  lastList: ?string
+): ?DraftBlockType {
+  if(tag === 'li') {
       if (lastList === 'ol') {
         return 'ordered-list-item';
       }
       return 'unordered-list-item';
-    case 'blockquote':
-      return 'blockquote';
-    case 'pre':
-      return 'code-block';
+  }
+
+  return null;
+}
+
+function getBlockMapSupportedTags(
+  draftBlockMap: DraftBlockMap
+): Array<DraftBlockType> {
+  const blockTypes = Object.keys(draftBlockMap);
+  const matchedTags = blockTypes.map(function(blockType) {
+      return draftBlockMap[blockType].element;
+  }).sort();
+
+  const supportedTags = matchedTags.filter(function(tag, index) {
+      const notUnstyledTag = tag !== draftBlockMap.unstyled.element;  // we do not want unstyled to be viewed as a valid match
+      const uniqueTag = matchedTags.indexOf(tag) === index;  // this will guarantee no duplicate tags added
+      return notUnstyledTag && uniqueTag;
+  });
+
+  return supportedTags;
+}
+
+// custom element conversions
+function getMultiMatchedType(
```

Comment:
This will be useful in the future, and we should think about how could we expose as an API, this would allow applications to define which blockTypes should be returned if there are multiple blockTypes for the same html tag, example 'iframe with internal and external url sources' or anything like that. We could expose a way to add functions to a set of transformations array